routing/endpointregistry.go

```diff
@@ -60,23 +73,54 @@ func (e *entry) SetLastSeen(ts time.Time) {
 	e.lastSeen.Store(ts)
 }
 
-func newEntry() *entry {
+func (e *entry) IncRequests(o IncRequestsOptions) {
+	curSlot := e.curSlot.Load()
+	e.totalRequests[curSlot].Add(1)
+	if o.FailedRoundTrip {
+		e.totalFailedRoundTrips[curSlot].Add(1)
+	}
```

Comment:
Such situation is possible, but has no critical circumstances: in this case statsUpdater will just calculate dropProbability ignoring a little amount of information from the very last requests in the period.
It may change the probability value a little bit, but it still will be reasonable and changed atomically.
Moveover, curSlot variable is read at the very beginning of this function, and it guarantees that all information about the given requests will go to the same slot.

Reply:
>  they can be changed by stats updater

Stats updater stores the data to the `nextSlot`, not the `curSlot`, which makes possibility of zeroing data request handler increments right now not practically possible.

Reply:
As discussed, we think the race might be ok